### PR TITLE
Support for logging tensor and numpy images

### PR DESCRIPTION
**Pull request by mecevit**
Support for logging the following as images:
- torch.Tensor
- numpy.array in `HWC`, `CHW` and `HW` formats

Example:

```python
img = np.zeros((100, 100, 3))
img[:, :, 0] = np.arange(0, 10000).reshape(100, 100) / 10000
img[:, :, 1] = 1 - np.arange(0, 10000).reshape(100, 100) / 10000

layer.log({"np_image": layer.Image(img, format="HWC")})
```

### CODE REVIEW

**Comment by fcakyon** [comment]
@mecevit tensorboard image constructor supports both float32 [0-1] and uint8 [0-255] images, is it the same in `layer.Image`?

from torch.utils.tensorboard source: https://github.com/pytorch/pytorch/blob/577e90ae9bf257040acb68da3626d9a64d07bf7a/torch/utils/tensorboard/summary.py#L431

**Comment by mecevit** [comment]
> @mecevit tensorboard image constructor supports both float32 [0-1] and uint8 [0-255] images, is it the same in `layer.Image`?
> 
> from torch.utils.tensorboard source: https://github.com/pytorch/pytorch/blob/577e90ae9bf257040acb68da3626d9a64d07bf7a/torch/utils/tensorboard/summary.py#L431

Good catch, addressed here: https://github.com/layerai/sdk/pull/153/commits/06efc2c0649b5a70988d11a66caf267ecad14ae6

image

File: layer/logged_data/image.py
Comment (cihati) [inline layer/logged_data/image.py:1]: Please move this code under layer/contracts/logged_data.py

File: layer/logged_data/image.py
Comment (cihati) [inline layer/logged_data/image.py:46]: `raise ValueError` if `format` is unsupported -- instead of doing it in `_get_image_from_array` below.

File: layer/main.py
Comment (cihati) [inline layer/main.py:751]: As we are not exposing `layer.Image`'s docs, this new PR's additions will not be very discoverable to the users. Either talk more about it here as part of `layer.log`, or make sure to send a PR for the docs separately in the monorepo.

Comment (mecevit) [reply layer/main.py:751]: I will create a PR in monorepo

File: layer/logged_data/image.py
Comment (mecevit) [reply layer/logged_data/image.py]: Done

File: layer/__init__.py
Comment (cihati) [inline layer/__init__.py:10]: Make sure to update this

**Review by cihati — APPROVED** [review]
Approving to unblock, but must address the last comment before merging.

File: layer/__init__.py
Comment (mecevit) [reply layer/__init__.py]: Good catch. I wonder why my unit tests didn't cover that. Fixed.

**Pull request merged**
